resources/views/cart/coupon.blade.php

```diff
@@ -1,32 +1,64 @@
+            >
+                <form v-on:submit.prevent="mutate" class="flex w-full flex-row gap-x-3">
+                    <x-rapidez::input
+                        :label="false"
                         class="text-ct-primary w-60"
                         name="couponCode"
-                        :placeholder="__('Enter code') . '...'"
-                        v-on="inputEvents"
-                        v-bind:value="couponCode"
+                        placeholder="Coupon code"
+                        v-model="variables.coupon_code"
                         v-bind:disabled="$root.loading"
                         required
                     />
-                    <x-rapidez-ct::button.outline type="submit" loader>
+                    <x-rapidez::button type="submit" class="sm:text-sm">
                         @lang('Apply')
-                    </x-rapidez-ct::button.outline>
-                </div>
-                <div v-if="cart.discount_name && cart.discount_amount < 0" class="text-ct-inactive mt-1 flex items-center gap-x-2">
-                    <button v-on:click="removeCoupon">
-                        <x-heroicon-s-x-mark class="h-4 w-4" />
-                    </button>
-                    @lang('Discount'): @{{ cart.discount_name }}
-                </div>
-                <div v-if="submitError" class="text-ct-error mt-1 italic">
-                    @{{ submitError }}
-                </div>
-            </form>
-        </coupon>
+                    </x-rapidez::button>
+                </form>
+            </graphql-mutation>
+
+            <div class="flex w-full flex-row gap-x-3">
+                <x-rapidez-ct::input
+                    class="text-ct-primary w-60"
+                    name="couponCode"
+                    :placeholder="__('Enter code') . '...'"
+                    v-on="inputEvents"
+                    v-bind:value="couponCode"
+                    v-bind:disabled="$root.loading"
+                    required
+                />
+                <x-rapidez-ct::button.outline type="submit" loader>
+                    @lang('Apply')
+                </x-rapidez-ct::button.outline>
+            </div>
+
+            <template v-if="cart.applied_coupons?.length" v-for="coupon in cart.applied_coupons" v-cloak>
+                <graphql-mutation
+                    :query="'mutation ($cart_id: String!) { removeCouponFromCart(input: { cart_id: $cart_id }) { cart { ' + config.queries.cart + ' } } }'"
+                    :variables="{ cart_id: mask }"
+                    :callback="updateCart"
+                    :error-callback="checkResponseForExpiredCart"
+                    v-slot="{ mutate }"
+                >
+                    <form v-on:submit.prevent="mutate" class="text-ct-inactive mt-1 flex items-center gap-x-2">
+                        <button type="submit">
+                            <x-heroicon-s-x-mark class="h-4 w-4 text-black-400"/>
```

Comment:
Can't we use a color from the checkout-theme? Like text-ct-inactive or something else?

Reply:
Also you can use size-4 instead of h-4 w-4